analyses/cell-type-neuroblastoma-04/final-annotation.Rmd

```diff
@@ -0,0 +1,571 @@
+  guides(color = guide_legend(override.aes = list(size = 2, alpha = 1))) +
+  umap_theme
+```
+
+#### Colored by PDX status
+
+```{r fig.width = 8, fig.height = 8}
+ggplot(umap_df) +
+  aes(x = UMAP1, y = UMAP2, color = is_xenograft) +
+  geom_point(alpha = 0.2, size = 0.3) +
+  scale_color_brewer(palette = "Set1") +
+  guides(color = guide_legend(override.aes = list(size = 2, alpha = 1))) +
+  umap_theme
+```
+
+
+### Dot plot
+
+This section shows dot plots of marker gene expression for the final annotations.
+Specifically, we show the top 5 highest `log2FC` marker genes per NBAtlas cell type for validation.
+Marker genes are taken directly from the NBAtlas paper where they were identified with `Seurat::FindMarkers()`. 
+
+Marker genes that are present in three or more cell types are not included.
+In addition, the upper limit for the color scale has been censored to a maximum of 6 to ensure visibility of expression levels across all cell types.
+
+```{r}
+input_dotplot_df <- final_annotation_df |>
+  dplyr::filter(final_label != "Unknown") |>
+  # recode cell types to match what is present in NBAtlas marker genes
+  dplyr::mutate(
+    label_recoded = dplyr::case_when(
+      stringr::str_detect(final_label, "monocyte") ~ "Monocyte", 
+      stringr::str_detect(final_label, "cDC") ~ "cDC",
+      .default = final_label
+  )) |>
+  dplyr::select(cell_id, label_recoded)
+
+# data frame of top 5 upregulated genes per cell type
+top_nbatlas_markers_df <- subset_nbatlas_markers(nbatlas_markers_df, 5, "up") |>
+  # only markers in no more than 2 categories
+  # this is a good middle ground to ensure there are at least 2 markers per cell type
+  dplyr::add_count(ensembl_gene_id) |>
+  dplyr::filter(n < 3) |>
+  dplyr::select(-n) 
+```
+
+
+```{r}
+# get total number of cells per final annotation group and set up y_label
+total_cells_df <- input_dotplot_df |>
+  dplyr::count(label_recoded, name = "total_cells") |>
+  dplyr::mutate(y_label = glue::glue("{label_recoded} ({total_cells})")) |>
+  dplyr::inner_join(label_map_df, by = c("label_recoded" = "NBAtlas_label")) |>
+  dplyr::group_by(NBAtlas_family) |>
+  dplyr::mutate(family_total_cells = sum(total_cells)) |>
+  # order by family count, but put Unknown at the end
+  dplyr::arrange(desc(family_total_cells)) |>
+  # amazing new dplyr trick; only this row is TRUE, so it gets moved to the last row
+  dplyr::arrange(label_recoded == "Unknown")
+
+total_cells_df$y_label <- factor(total_cells_df$y_label, levels = rev(total_cells_df$y_label))
+nbatlas_bar_order <- total_cells_df$label_recoded
+```
+
+```{r}
+# Prepare the expressed_genes vector
+# we only care about if that gene is expressed otherwise we won't waste memory and include it
+gene_sums <- rowData(merged_sce) |>
+  as.data.frame() |>
+  dplyr::select(contains("detected")) |>
+  as.matrix() |>
+  rowSums()
+expressed_genes <- names(gene_sums)[gene_sums > 0]
+```
+
+```{r, fig.width = 32, fig.height = 16, eval = !params$testing}
```

Comment:
Yikes this is some really small font! I feel like we could maybe increase the font size for the labels here?